src/probnum/linops/_linear_operator.py

```diff
@@ -539,52 +895,128 @@ def __init__(
         self,
         A: Union[np.ndarray, scipy.sparse.spmatrix],
     ):
-        super().__init__(A=A)
+        if isinstance(A, scipy.sparse.spmatrix):
+            self.A = A
 
-    def _matvec(self, x):
-        return self.A @ x  # Needed to call __matmul__ instead of np.dot or np.matmul
+            shape = self.A.shape
+            dtype = self.A.dtype
 
-    def _rmatvec(self, x):
-        return np.conj(self.A.T @ np.conj(x))
+            matmul = LinearOperator.broadcast_matmat(lambda x: self.A @ x)
+            rmatmul = LinearOperator.broadcast_rmatmat(lambda x: x @ self.A)
+            todense = self.A.toarray
+            inverse = lambda: Matrix(scipy.sparse.linalg.inv(self.A))
+            trace = lambda: self.A.diagonal().sum()
+        else:
+            self.A = np.asarray(A)
 
-    def _matmat(self, X):
-        return self.A @ X
+            shape = self.A.shape
+            dtype = self.A.dtype
 
-    def todense(self):
-        if isinstance(self.A, scipy.sparse.spmatrix):
-            return self.A.toarray()
+            matmul = lambda x: A @ x
+            rmatmul = lambda x: x @ A
+            todense = lambda: self.A
+            inverse = None
```

Comment:
Is there a reason for implementing the inverse for sparse matrices but not dense matrices beyond the cubic complexity?

Reply:
The fallback inverse, i.e. `_InverseLinearOperator`, implements a "lazy" inverse computation via LU decomposition  which is in general faster than naively inverting and since we should hardly ever need the full inverse matrix I preferred this over the naive inversion. Any thoughts?